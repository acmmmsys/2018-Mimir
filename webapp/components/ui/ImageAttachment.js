import React, { Component } from 'react';
import classnames from 'classnames';

import 'style/ui/ImageAttachment.scss'

const defaultClasses = [
    'ui-image-attachment-container',
];

class ImageAttachment extends Component {

    static defaultProps = {
        classNames: []
    }

    render() {

        const classNames = classnames(
            defaultClasses,
            this.props.classNames,
        );

        return (
            <div className={classNames}>
                <span className='ui-image-attachment-label'>
                    { this.props.label }
                </span>
                <img className='ui-image-attachment' src={this.props.src} /> 
            </div>
        );
    }
}

export default ImageAttachment;