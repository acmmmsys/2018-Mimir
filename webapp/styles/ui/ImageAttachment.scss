

.ui-image-attachment {
    max-width: 120px;
    max-height: 120px;
}

.ui-image-attachment-container {
    max-width: 120px;
    max-height: 120px;
}

.ui-image-attachment-label {
    width: 100%;
    text-align: left;
    padding: 1px 10px;
    white-space: nowrap;
    overflow: hidden;
    display:block;
    text-overflow: ellipsis;
}