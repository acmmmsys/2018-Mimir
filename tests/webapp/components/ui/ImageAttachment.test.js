import React from 'react';
import ImageAttachment from '../../../../webapp/components/ui/ImageAttachment'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('ImageAttachment does render', () => {
  const wrapper = shallow(
      <ImageAttachment store={mockStore(initialState)}>This is a test!</ImageAttachment>
  );
  expect(wrapper).toMatchSnapshot();
});
