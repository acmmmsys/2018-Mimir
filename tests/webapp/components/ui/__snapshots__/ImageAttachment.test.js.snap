// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageAttachment does render 1`] = `
<div
  className="ui-image-attachment-container "
>
  <span
    className="ui-image-attachment-label"
  />
  <img
    className="ui-image-attachment"
  />
</div>
`;
